src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,189 @@ void VulnerabilityScannerFacade::start(
+                                              ? syncMsg->data_as_state()
+                                                    ->attributes_as_syscollector_packages()
+                                                    ->name()
+                                                    ->c_str()
+                                              : "");
+                                scanOrchestrator->run(data);
+                            }
+                            catch (const std::exception& e)
+                            {
+                                logWarn(WM_VULNSCAN_LOGTAG,
+                                        "Discarded syscollector_packages event: %s package. %s",
+                                        (syncMsg->data_as_state() &&
+                                         syncMsg->data_as_state()->attributes_as_syscollector_packages() &&
+                                         syncMsg->data_as_state()->attributes_as_syscollector_packages()->name())
+                                            ? syncMsg->data_as_state()
+                                                  ->attributes_as_syscollector_packages()
+                                                  ->name()
+                                                  ->c_str()
+                                            : "",
+                                        e.what());
+                            }
+                        }
+                    }
+                }
+            });
+
         // Subscription to syscollector delta events.
         m_syscollectorDeltasSubscription =
             std::make_unique<RouterSubscriber>("deltas-syscollector", "vulnerability_scanner_deltas");
         m_syscollectorDeltasSubscription->subscribe(
-            [scanOrchestrator](const std::vector<char>& message)
+            [scanOrchestrator, this](const std::vector<char>& message)
             {
+                const SyscollectorDeltas::Delta* delta = SyscollectorDeltas::GetDelta(message.data());
                 std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*> data =
-                    SyscollectorDeltas::GetDelta(message.data());
+                    delta;
 
-                scanOrchestrator->run(data);
+                try
+                {
+                    scanOrchestrator->run(data);
+                }
+                catch (const OsDataException& e)
+                {
+                    m_deltaQueue->push(std::move(std::string(message.data(), message.size())));
+                    logDebug1(WM_VULNSCAN_LOGTAG,
+                              "%s. dbsync_packages events postponed: %s package. Queue size: %d",
+                              e.what(),
+                              (delta->data_as_dbsync_packages() && delta->data_as_dbsync_packages()->name())
+                                  ? delta->data_as_dbsync_packages()->name()->c_str()
+                                  : "",
```

Comment:
This expression is really hard to follow, can we declare it as a const variable before the debug call with a descriptive name?

Reply:
Ditto